spring-rest2ts-generator/src/main/java/com/blueveery/springrest2ts/tsmodel/TSComplexElement.java

```diff
@@ -29,11 +30,21 @@ public TSComplexElement(String name, TSModule module, ImplementationGenerator im
         this.implementationGenerator = implementationGenerator;
     }
 
+    public TSComplexElement(
+            String name, TSModule module, ImplementationGenerator implementationGenerator, TSFormalTypeParameter... formalTypeParameters
+    ) {
+        this(name, module, implementationGenerator);
+        tsFormalTypeParameterList.addAll(Arrays.asList(formalTypeParameters));
+    }
 
     public SortedSet<TSField> getTsFields() {
         return tsFields;
     }
 
+     public TSField getFieldByName(String name) {
+        return tsFields.stream().filter(f -> name.equals(f.getName())).findFirst().orElse(null);
```

Comment:
Maybe a better idea could be to change the signature to Optional<TSField> instead returning null

Reply:
In the entire TS model API we don't use Optional so this change wouldn't be consistent and for sure we will not migrate model to use Optional